epoch/api/naming_system_api_usage.md

```diff
@@ -0,0 +1,95 @@
+[back](./README.md)
+# Naming System - intended usage
+
+## Introduction
+
+The way to interact with Naming System is to use HTTP API.
+To read about possible states and life cycle of a name please see [the doc](/AENS.md).
+
+The following assumes that the node exposes at address 127.0.0.1 the following ports:
+* HTTP external API: 3013
+* HTTP internal API: 3113
+
+## Name management flow
+
+Check the public key of the node first (used as account key in later API calls):
+```
+curl http://localhost:3113/v2/account/pub-key
+{"pub_key":"ak$3scLu3oJbhsdCJkDjfJ6BUPJ4M9ZZJe57CQ56deSbEXhaTSfG3Wf3i2GYZV6APX7RDDVk4Weewb7oLePte3H3QdBw4rMZw"}
+```
+
+Verify that the name that your going to claim is not taken.
+For the purpose of this example let's consider `fooo.baar.aet` name:
+```
+curl http://localhost:3013/v2/name\?name\=fooo.baar.aet
+{"reason":"Name not found"}
+```
+
+In order to claim a name you need to submit a preclaim transaction first, containing commitment hash.
+
+You need to pick a random salt value, to make commitment hash unique.
+Commitment hash is calculated based on [the formula](/AENS.md#pre-claim).
+For the purpose of this example name `fooo.baar.aet` with salt 123 is considered:
+```
+Base58(Commitment(fooo.baar.aet, 123)) = "cm$2fxgad7CSY4dPsn6SVtgdigNq8gdzRJb9W3VB1W5oRKHqC3yH6"
```

Comment:
@sennui @zp-sd  Would it make sense to provide a `bin/epoch eval ...` snippet for this release? (`eval` or `rpcterms` - not sure which one is better) so that user can reproduce quickly without rewriting formula?

Reply:
```
$ bin/epoch eval 'aec_base58c:encode(commitment, aens_hash:commitment_hash(<<"fooo.baar.aet">>, 123))'
<<"cm$2fxgad7CSY4dPsn6SVtgdigNq8gdzRJb9W3VB1W5oRKHqC3yH6">>
```